torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -383,8 +383,12 @@ def wrapper(
             self.device_type = "cpu"
 
         self.init_pg()
-        func(self, *args, **kwargs)  # type: ignore[misc]
-        self.destroy_pg()
+        try:
+            func(self, *args, **kwargs)  # type: ignore[misc]
+        except Exception as e:
+            raise e
+        finally:
+            self.destroy_pg()
```

Comment:
It looks like `self.destroy_pg()` is:
```
dist.barrier()
dist.destroy_process_group()
```

IIUC, `func()` is being run in the calling process and does not spawn any new process or thread. What happens if one rank raises an exception but another rank does not? Would the `barrier()` hang in `self.destroy_pg()`?

(For context, I want to get rid of this warning for FSDP tests too, but I was not sure if this approach will always work.)

Reply:
Ideally, we should not have to call `barrier()` before calling `destroy_pg()`, but I'm not sure if that is the case. cc., @wconstab . However, this version is still better than the original one. If a rank raises while other don't, the original code would have the same issue that one rank doesn't participate in the `barrier()` call.